### PR TITLE
Add initial draft of e2e scripts

### PR DESCRIPTION
**Pull request by ChrisJar**

### CODE REVIEW

File: e2e-cat.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+cluster = LocalCUDACluster(
+    n_workers=1,
+    CUDA_VISIBLE_DEVICES="2",
+    protocol="ucx",
+    jit_unspill=True,
+    rmm_pool_size="70GiB",
+    device_memory_limit="78GB",
+)
+client = Client(cluster)
```
Comment (VibhuJawa) [inline e2e-cat.py:23]: I dont think this will work. You need this inside __main__

File: e2e-cat.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import cudf
```
Comment (VibhuJawa) [inline e2e-cat.py:1]: Can you also add a readme with the current results

File: e2e-cat.py [hunk 3]
```diff
@@ -12,64 +12,68 @@
...
+    )
+    t1 = time.time()
+    print("Training Time: {}".format(t1-t0))
+
+    del dtrain
+
+    test_ddf = dd.read_parquet(test_data_dir, split_row_groups=True)
+
+    test_ddf = transform_df(test_ddf)
+    X_test, y_test = test_ddf.drop("label", axis=1), test_ddf["label"]
```
Comment (trivialfis) [inline e2e-cat.py:65]: `drop` might cause a dataframe copy.

**Review by trivialfis — COMMENTED** [review]
Hi, from the document you wrote, it seems memory usage is the priority. Have you considered splitting up the ETL process with the training process to account for memory usage for each phrase?

```
ETL -> cache parquet -> training
```
We can get an estimate for each different phrase and identify the source of memory consumption.

**Pull request closed without merging**